### PR TITLE
Aslr pager

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
This is the final pull request in the ASLR series. With this ASLR works with paging enabled.

File: core/arch/arm/kernel/generic_boot.c [hunk 1]
```diff
@@ -436,7 +494,7 @@ static void init_runtime(unsigned long pageable_part)
 	 * Linear memory (flat map core memory) ends there.
 	 */
 	mm = tee_mm_alloc2(&tee_mm_vcore, VCORE_UNPG_RX_PA,
-			   (vaddr_t)(__pageable_start - VCORE_UNPG_RX_PA));
+			   (vaddr_t)(__pageable_start - VCORE_UNPG_RX_PA) - 1);
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/arch/arm/kernel/generic_boot.c]: This seems to have been added by mistake, I'll remove it.

**Comment by jenswi-linaro** [comment]
Tags applied. Note that in the final commit "Enable ASLR by default" both plat-vexpress and plat-hikey are updated. I've added another fix for the pager too, "core: fix tee_pager_release_one_phys() assert".

It seems that `CFG_CORE_ASLR=y` can trigger a few corner cases. I think it's good to have it enabled by default now in order to give us some time to smoke out eventual problems before the next release.

**Comment by jenswi-linaro** [comment]
Update

**Comment by jforissier** [comment]
For commit "core: fix tee_pager_release_one_phys() assert":

`Acked-by: Jerome Forissier <jerome@forissier.org>`

For commit "core: link.mk: Make sure to link without relro":

`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by jenswi-linaro** [comment]
Tags applied. I added as fixes tag to "core: fix pager vaspace start in assign_mem_va()" also.

**Comment by jforissier** [comment]
Not sure why IBART reports "Job cancelled!" -- the tests seem to have run fine. I am also ignoring the checkpatch error on `KEEP_PAGER()`. As mentioned [in another PR](https://github.com/OP-TEE/optee_os/pull/3377#issuecomment-564929705) we should perhaps rename this macro one day.

**Pull request merged**